ChangeLog.d/8358.txt

```diff
@@ -0,0 +1,9 @@
+Features
+   * If a cipher or AEAD mechanism has a PSA driver, you can now build the
+     library without the corresponding built-in implementation and without
+     enabling MBEDTLS_CIPHER_C.
```

Comment:
I think we should leave `CIPHER_C` out of this entry, it already has its own.

Also, I'd add "see docs/driver-only-builds.md for full details and current limitations" because there are some limitations.